### DISCUSSION TITLE
CSLA 6 prerelease 4 online

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#2728] [Announcements]
I am pleased to announce that CSLA 6 prerelease 4 is available in nuget.

This prerelease includes some important fixes and changes based on your feedback from prerelease 3. Nothing replaces your input into this process, and I am deeply grateful for those of you who are taking time to provide feedback.

* Fixed an issue where more than one DI scope was being created for logical server-side processing
* FIxed issue with lazy loaded properties
* Lazy loaded properties can now access an `IDataPortal<T>` via the new protected `GetDataPortal` method
* Add `RegisterContextManager` method in config
* Normalized some config API irregularities
* Blazor now has its own application context manager that works with Blazor authentication
* Business and authorization rule context objects now have a `GetDataPortal` method to make it possible for a rule to invoke the data portal
* Implement a new interceptor capability in the server-side data portal, where it is possible to register multiple interceptors that are invoked before/after a data portal operation method is called - you can use this for many purposes such as logging or exception handling
* Implement an interceptor that sanitizes server-side exceptions, removing potentially sensitive information that shouldn't be returned to an unknown client in the case of an exception

Many thanks to

* @TheCakeMonster 
* @peranborkett 
* @wilfreds
* @swegele 
* @mtavares628
* @kcabral817

and probably some other folks that I've missed - your PRs, comments, and feedback have been so useful!

**Comment by michaelcsikos** [comment]
Hi @rockfordlhotka I'm just wondering if you have a best- and worst-case estimate of when you think CSLA 6 will be ready for prime time? I'm hoping to start a new Blazor project in the near future and would prefer to go straight to .NET 6.0 if possible. Blazor is still quite new for me and I am re-reading *Using CSLA 5: Blazor and WebAssembly* presently.

**Comment by rockfordlhotka** [comment]
@michaelcsikos today I finished what I believe to be the final work item for CSLA 6, other than updating the samples and ProjectTracker - which might reveal other items that need resolution.

That said, the Blazor, MVC, Razor Pages, WPF, and Windows Forms samples were already updated for previous prereleases, so I don't expect major surprises from them.

So we're in the end game with CSLA 6 at this point. I'd absolutely recommend you use CSLA 6, _especially_ for server-side Blazor (since CSLA 5 has a major flaw), but also for Blazor wasm.
